tests/cephfs/cephfs_mirroring/cephfs_mirroring_utils.py

```diff
@@ -1329,3 +1331,424 @@ def validate_snaps_status_increment(self, json_before, json_after, snap_status):
+        :param dir_path: Directory where files should be created
+        :param num_files: Number of files to create
+        :param size_gb: Size of each file in GB
+        """
+        generate_file_script = "generate_files_for_snapdiff.py"
+        remote_path = f"/root/{generate_file_script}"
+
+        client.upload_file(
+            sudo=True,
+            src=f"tests/cephfs/lib/{generate_file_script}",
+            dst=remote_path,
+        )
+
+        client.exec_command(
+            sudo=True,
+            cmd=f"python3 {remote_path} {dir_path} {num_files} {size_gb}",
+            timeout=14400,
+        )
+
+    def modify_files_for_snapdiff(
+        self, client, dir_path, num_files, mode="write", length=5, bytes_size=None
+    ):
+        """
+        Uploads and executes the modify_file_at_10_random_offsets.py script on the remote client.
+
+        :param client: Remote client object
+        :param dir_path: Directory where files should be created or read
+        :param num_files: Number of files to modify
+        :param mode: 'write' or 'read'
+        :param length: Number of random offsets
+        :param bytes_size: Required only for write mode, size of data to write at each offset (e.g., '1M', '512K')
+        """
+        modify_script = "modify_file_at_10_random_offsets.py"
+        remote_path = f"/root/{modify_script}"
+
+        client.upload_file(
+            sudo=True,
+            src=f"tests/cephfs/lib/{modify_script}",
+            dst=remote_path,
+        )
+
+        cmd = f"python3 {remote_path} {dir_path} --file-count {num_files} --mode {mode} --length {length}"
+        if mode == "write" and bytes_size:
+            cmd += f" --bytes {bytes_size}"
+
+        client.exec_command(
+            sudo=True,
+            cmd=cmd,
+            timeout=14400,
+        )
+
+    def create_snapshot_snapdiff(
+        self,
+        client,
+        mounting_dir,
+        subvol_path,
+        snap_name,
+        source_fs,
+        subvolume=False,
+        subvol_name=None,
+        subvol_group=None,
+    ):
+        """
+        Function to create a snapshot (either path-based or subvolume-based).
+        """
+        if subvolume:
+            if not subvol_name or not subvol_group:
+                raise ValueError(
+                    "subvol_name and subvol_group are required when subvolume=True"
+                )
+
+            cmd = (
+                f"ceph fs subvolume snapshot create {source_fs} {subvol_name} "
+                f"{snap_name} {subvol_group}"
+            )
+            log.info(f"Creating subvolume snapshot using CLI: {cmd}")
+            client.exec_command(sudo=True, cmd=cmd)
+        else:
+            # Default path-based snapshot creation
+            client.exec_command(
+                sudo=True, cmd=f"mkdir {mounting_dir}{subvol_path}/.snap/{snap_name}"
+            )
+
+    # Function to validate snapshot sync with retry
+    @retry(CommandFailed, tries=10, delay=60)
```

Comment:
Hi Hemanth, 
This retry mechanism runs for 17 hours. Can we consider reducing the delay value? If the system enters a bad state, the test case could remain stuck for the entire duration

Reply:
Updated.

Reply:
Is the retry added because the function/sync does not report consistently? And do we really need to try 10 times?